services/rule_engine/src/main/java/agents/Smith.java

```diff
@@ -1,27 +1,77 @@
 package agents;
 
 import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
 
+import com.mindsmiths.gsheetsAdapter.GSheetsAdapterAPI;
 import com.mindsmiths.gsheetsAdapter.reply.Spreadsheet;
 import com.mindsmiths.ruleEngine.model.Agent;
+import com.mindsmiths.ruleEngine.util.Agents;
 import com.mindsmiths.ruleEngine.util.Log;
 
+
 import lombok.AllArgsConstructor;
 import lombok.Data;
 import lombok.ToString;
 
+
 @Data
 @ToString(callSuper = true)
 @AllArgsConstructor
 public class Smith extends Agent {
 
     LocalDateTime lastTableUpdate = null;
+    int nextMailRow = 0;
+    int nextAnsRow = 0;
+    Spreadsheet sheet;
 
+    // Agent id
     public Smith() {
         this.id = "SMITH";
     }
 
-    public void processSheet(Spreadsheet spreadsheet){
-        Log.info(spreadsheet);
+    // Getting the sheet as an object and getting the first free row in the spreadsheet
+    public void processSpreadsheet(Spreadsheet spreadsheet){
+        sheet = spreadsheet;
+        List<Map<String, String>> emails = spreadsheet.getSheets().get("Mailovi");
+        List<Map<String, String>> answers = spreadsheet.getSheets().get("Odgovori");
+        nextMailRow = emails.size() + 2;
+        nextAnsRow = answers.size() + 2;
+    }
+
+    /* 
+
+    OPTIONAL - Function which updates the "Mailovi" sheet with a mock mail for now
+
+    public void updateGSheet() {
+        List<List<String>> values = List.of(List.of("jedan@mail.com"));
+        String range = String.format("Mailovi!A%d",nextRow+2);
+        GSheetsAdapterAPI.updateSheet(values, range);
+    }
+
+    */
+
+    // Function to update "Odgovori" sheet with given parameters
+    // TODO: set the function up to work with data collected from armory
+    public void getReview(String mail, String grade, String review){
+        String range = String.format("Odgovori!A%d:C%d", nextAnsRow, nextAnsRow);
+        List<List<String>> values = List.of(List.of(mail,ocjena,review));
+        GSheetsAdapterAPI.updateSheet(values, range);
+    }
+
+    // Going through the list of mails in the spreadsheet and creating agents for mails who don't already have one
+    public void createNewAgents(){
+        List<Map<String, String>> mailList = sheet.getSheets().get("Mailovi");
+
+        for (Map<String, String> item : mailList) {
+            for (String key : item.keySet()) {
+                String email = item.get(key);
+                if(!Agents.exists(email)){
```

Comment:
This won't work because `Agents.exists()` takes the agent's id as its parameter. The user's email is not the agent's id (the id is a string of letters and numbers and is randomly generated).

Reply:
We are putting the mail as an id so we can more easily differentiate them